x/blob/types/payforblob.go

```diff
@@ -331,6 +332,51 @@ func ValidateBlobs(blobs ...*Blob) error {
 	return nil
 }
 
+// validateBlobData returns an error if the size of data is zero or too large.
+func validateBlobData(appVersion uint64, data []byte) error {
+	if len(data) == 0 {
+		return ErrZeroBlobSize
+	}
+
+	maxBlobSize := blobSizeUpperBound(appVersion)
+	if len(data) > maxBlobSize {
+		return ErrBlobSizeTooLarge.Wrapf("max blob size is %d bytes", maxBlobSize)
```

Comment:
🤔 the wrapped error should probably include the blob size for debugging purposes